tokens/src/lib.rs

```diff
@@ -288,29 +306,107 @@ decl_error! {
 }
 
 impl<T: Trait> Module<T> {
+	/// Check whether account_id is a module account
+	fn is_module_account_id(account_id: T::AccountId) -> bool {
+		let account: AccountId32 = Into::<AccountId32>::into(Into::<T::AccountIdConvert>::into(account_id));
+		let data: [u8; 32] = account.into();
+		data.starts_with(&ModuleId::TYPE_ID)
+	}
+
+	fn existential_deposit(currency_id: T::CurrencyId) -> T::Balance {
+		T::ExistenceDeposits::get()
+			.iter()
+			.find(|&&x| x.0 == currency_id)
+			.map(|a| a.1)
+			.unwrap_or_else(Zero::zero)
+	}
+
+	fn post_account_mutation(
+		who: &T::AccountId,
+		currency_id: T::CurrencyId,
+		new: AccountData<T::Balance>,
+		existed: bool,
+	) -> Option<AccountData<T::Balance>> {
+		let total = new.total();
+		if total < Self::existential_deposit(currency_id) && !Self::is_module_account_id(who.clone()) {
+			// remove dust
+			if !total.is_zero() {
+				TotalIssuance::<T>::mutate(currency_id, |t| {
```

Comment:
Shouldn't modify `TotalIssuance` here. It is up to OnDust to decide if we want to burn the token (modify `TotalIssuance` or transfer to treasury or doing something else.

Reply:
`TotalInsurance` is still deducted here to avoid interface refactoring